packages/graph/tests/votingStrategy/quadraticFunding/implementation.test.ts

```diff
@@ -0,0 +1,160 @@
+import { test, assert, newMockEvent, createMockedFunction, describe, beforeEach, clearStore, afterEach, logStore } from "matchstick-as/assembly/index";
+import { Address, BigInt, ethereum, log } from "@graphprotocol/graph-ts";
+import { handleVote } from "../../../src/votingStrategy/quadraticFunding/implementation";
+import { Voted as VotedEvent } from "../../../generated/QuadraticFundingVotingStrategy/QuadraticFundingVotingStrategyImplementation";
+import { QFVote, Round, VotingStrategy } from "../../../generated/schema";
+
+
+let token: Address;
+let amount: BigInt;
+let voter: Address;
+let grantAddress: Address;
+let roundAddress: Address;
+
+let newVoteEvent: VotedEvent;
+
+let votingStrategyAddress: Address;
+
+function createNewVotedEvent(
+  token: Address,
+  amount: BigInt,
+  voter: Address,
+  grantAddress: Address,
+  roundAddress: Address
+): VotedEvent {
+  const newVoteEvent = changetype<VotedEvent>(newMockEvent());
+
+  const tokenParam = new ethereum.EventParam("token", ethereum.Value.fromAddress(token));
+  const amountParam = new ethereum.EventParam("amount", ethereum.Value.fromUnsignedBigInt(amount));
+  const voterParam = new ethereum.EventParam("voter", ethereum.Value.fromAddress(voter));
+  const grantAddressParam = new ethereum.EventParam("grantAddress", ethereum.Value.fromAddress(grantAddress));
+  const roundAddressParam = new ethereum.EventParam("roundAddress", ethereum.Value.fromAddress(roundAddress));
+
+  newVoteEvent.parameters.push(tokenParam);
+  newVoteEvent.parameters.push(amountParam);
+  newVoteEvent.parameters.push(voterParam);
+  newVoteEvent.parameters.push(grantAddressParam);
+  newVoteEvent.parameters.push(roundAddressParam);
+
+
+  // TODO: how to set event contract address as votingStrategy contract
+  // We need to be able to set event.receipt.logs[0].address as the VotingStrategy
+  // Only then can the handler set the QFVote.votingStrategy value
```

Comment:
This gets tricky cause ideally  we wanna be able to set the contract address deploying the event manually